File: scripts/reorganize-realestate-page.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, '..', 'src', 'app', 'real-estate-companies-in-kenya', 'RealEstateCompaniesPage.tsx');

// Read file
let content = fs.readFileSync(filePath, 'utf8');

// Find line numbers for both sections
const lines = content.split('\n');
let overviewStart = -1;
let overviewEnd = -1;
let companiesStart = -1;
let companiesEnd = -1;

// Find overview section
for (let i = 0; i < lines.length; i++) {
  if (lines[i].includes('id="overview"') && lines[i].includes('<section')) {
    overviewStart = i;
  }
  if (overviewStart !== -1 && overviewEnd === -1 && lines[i].trim().startsWith('</section>')) {
    overviewEnd = i;
    break;
  }
}

// Find companies section
for (let i = 0; i < lines.length; i++) {
  if (lines[i].includes('id="companies"') && lines[i].includes('<section')) {
    companiesStart = i;
  }
  if (companiesStart !== -1 && companiesEnd === -1 && i > companiesStart && lines[i].trim().startsWith('</section>')) {
    companiesEnd = i;
    break;
  }
}

console.log('Overview section:', overviewStart, '-', overviewEnd);
console.log('Companies section:', companiesStart, '-', companiesEnd);

// Extract sections
const overviewSection = lines.slice(overviewStart, overviewEnd + 1);
const companiesSection = lines.slice(companiesStart, companiesEnd + 1);

// Remove companies section
lines.splice(companiesStart, companiesEnd - companiesStart + 1);

// Insert companies section before overview (accounting for removed lines)
const newOverviewStart = overviewStart > companiesStart ? overviewStart - (companiesEnd - companiesStart + 1) : overviewStart;
lines.splice(newOverviewStart, 0, ...companiesSection, '');

// Join back
content = lines.join('\n');

// Now improve colors and contrast
// Change blue/purple gradient to green
content = content.replace(
  /bg-gradient-to-r from-blue-600 via-indigo-600 to-purple-600/g,
  'bg-gradient-to-r from-green-700 via-green-600 to-teal-600'
);

// In companies section, change button text color
content = content.replace(
  /text-blue-600 px-6 py-3 rounded-lg hover:bg-blue-50/,
  'text-green-700 px-6 py-3 rounded-lg hover:bg-green-50'
);

// Change text-blue-100 to text-green-100 in companies section
content = content.replace(/text-blue-100/g, 'text-green-100');

// Reduce yellow, change to orange/amber
content = content.replace(/from-yellow-50 to-yellow-100/g, 'from-orange-50 to-amber-50');
content = content.replace(/border-yellow-200/g, 'border-orange-200');
content = content.replace(/text-yellow-600/g, 'text-orange-600');

// Write back
fs.writeFileSync(filePath, content, 'utf8');

console.log('Successfully reorganized Real Estate Companies page!');
console.log('  - Companies section moved before Market Overview');
console.log('  - Color scheme improved (green/professional theme)');
console.log('  - Better visual contrast');

File: scripts/reorganize-realestate-page.py
#!/usr/bin/env python3
"""
Reorganize Real Estate Companies page:
1. Move companies section to appear before market overview
2. Improve color scheme and contrast
3. Make search bar more visible
"""

import re

# Read the file
with open('src/app/real-estate-companies-in-kenya/RealEstateCompaniesPage.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

# Find the companies section (starts at line ~1197)
companies_pattern = r'(<section id="companies".*?</section>)\s*(?=\s*{/\* Related Resources Section \*/)'
companies_match = re.search(companies_pattern, content, re.DOTALL)

if not companies_match:
    print("Could not find companies section!")
    exit(1)

companies_section = companies_match.group(1)

# Remove companies section from its current location
content_without_companies = content[:companies_match.start()] + content[companies_match.end():]

# Find where to insert (right after "div className="container mx-auto px-4 py-12">")
insertion_point = content_without_companies.find('<div className="container mx-auto px-4 py-12">')
if insertion_point == -1:
    print("Could not find insertion point!")
    exit(1)

# Find the end of that opening div tag line
insertion_point = content_without_companies.find('\n', insertion_point) + 1

# Insert companies section at the top
new_content = (
    content_without_companies[:insertion_point] +
    '\n          ' + companies_section + '\n\n' +
    content_without_companies[insertion_point:]
)

# Now improve the visual design
# 1. Change blue gradient to green for companies header
new_content = new_content.replace(
    'bg-gradient-to-r from-blue-600 via-indigo-600 to-purple-600',
    'bg-gradient-to-r from-green-700 via-green-600 to-teal-600'
)

# 2. Change button colors from blue to green
new_content = re.sub(
    r'text-white text-blue-600',
    'text-white text-green-700',
    new_content
)
new_content = re.sub(
    r'hover:bg-blue-50',
    'hover:bg-green-50',
    new_content, count=2  # Only first few occurrences in companies section
)

# 3. Change text colors in companies section from blue to green
new_content = new_content.replace('text-blue-100', 'text-green-100')

# 4. Improve statistics cards - reduce yellow, use more neutral colors
new_content = new_content.replace(
    'bg-gradient-to-br from-yellow-50 to-yellow-100 rounded-xl p-4 border-2 border-yellow-200',
    'bg-gradient-to-br from-orange-50 to-amber-50 rounded-xl p-4 border-2 border-orange-200'
)
new_content = new_content.replace(
    'text-yellow-600',
    'text-orange-600'
)

# Write the reorganized file
with open('src/app/real-estate-companies-in-kenya/RealEstateCompaniesPage.tsx', 'w', encoding='utf-8') as f:
    f.write(new_content)

print("Successfully reorganized Real Estate Companies page!")
print("   - Companies section moved to top (before Market Overview)")
print("   - Color scheme improved (green theme)")
print("   - Better contrast and professional appearance")
